File: CUSTOR.EICOnline.DAL/EntityLayer/Investor/Investor.cs
using CUSTOR.EICOnline.DAL.Enum;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CUSTOR.EICOnline.DAL.EntityLayer
{
    public partial class Investor : IAuditableEntity
    {
        public Investor()
        {
            //Associate = new HashSet<Associate>();

        }

        public virtual string InvestorName
        {
            get
            {
                string investorName = FirstName + ' ' + FatherName + ' ' + GrandName;
                return investorName;
            }
        }

        public virtual string InvestorNameEng
        {
            get
            {
                string investorName = FirstNameEng + ' ' + FatherNameEng + ' ' + GrandNameEng;
                return investorName;
            }
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int InvestorId { get; set; }
        //public int AddressId { get; set; }
        public int LegalStatus { get; set; }

        public int? Title { get; set; }
        public string FirstName { get; set; }

        public string FirstNameSort { get; set; }
        public string FirstNameSoundx { get; set; }
        public string FirstNameEng { get; set; }
        public string FatherName { get; set; }
        public string FatherNameSort { get; set; }
        public string FatherNameSoundx { get; set; }
        public string FatherNameEng { get; set; }
        public string GrandName { get; set; }
        public string GrandNameSort { get; set; }
        public string GrandNameSoundx { get; set; }
        public string GrandNameEng { get; set; }

        //public DateTime? DateOfBirth { get; set; }
        public int? Gender { get; set; }
        public int? FormOfOwnership { get; set; }
        public int? Nationality { get; set; }
        public bool IsDiaspora { get; set; }
        public bool IsEthiopianOrigin { get; set; }

        //public byte?[] Photo { get; set; }
        public string Tin { get; set; }

        public string RegistrationNumber { get; set; }
        public DateTime? RegistrationDate { get; set; }
        public string TradeName { get; set; }
        public string TradeNameSort { get; set; }
        public string TradeNameSoundX { get; set; }
        public string TradeNameEnglish { get; set; }
        public string UserId { get; set; }
        //public string SiteCode { get; set; }
        public bool? IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public string CreatedUserId { get; set; }
        public string UpdatedUserId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string UpdatedBy { get; set; }

        public Guid ObjectId { get; set; }

    }
}

File: CUSTOR.EIC.API/src/app/model/investor.ts
import {Injectable} from '@angular/core';
import {Validators} from '@angular/forms';

@Injectable()
export class Investor {
  InvestorName?: string = null;
  InvestorNameEng?: string = null;
  TradeName?: string;
  TradeNameEnglish?: string;
  InvestorId = 0;
  LegalStatus? = 0;
  Title: number;
  FormOfOwnership?: number;
  FirstName: string = null;
  // firstNameSort: string = null;
  // firstNameSoundx: string = null;
  FirstNameEng: string = null;
  FatherName: string = null;
  // fatherNameSort: string = null;
  // fatherNameSoundx: string = null;
  FatherNameEng: string = null;
  GrandName: string = null;
  // grandNameSort: string = null;
  // grandNameSoundx: string = null;
  GrandNameEng: string = null;
  Gender: string = null;
  Nationality: string = null;
  BranchCountry: string = null;
  IsEthiopianOrigin = false;
  // IsDiaspora = false;
  Tin?: string = null;
  RegistrationNumber?: string = null;
  RegistrationCatagories: string[];
  RegistrationDate?: Date = null;
  // TradeName?: string = null;
  // tradeNameSort: string = null;
  // tradeNameSoundX: string = null;
  // TradeNameEnglish?: string = null;

  PaidCapital = null;
  SighnedCapital = null;

  UserId?: string = null;
  IsExistingCustomer: boolean;
  // SiteCode: string = null;
  IsActive = true;
  IsDeleted = false;
  RegionId: string = null;
  ZoneId: string = null;
  WoredaId: string = null;
  WoredaEngId: string = null;
  KebeleId: string = null;
  KebeleEngId: string = null;
  HouseNo: string = null;
  TeleNo: string = null;
  Pobox: string = null;
  Fax: string = null;
  CellPhoneNo: string = null;
  Email: string = null;
  OtherAddress: string = null;
  UserName: string = null;
  AddressId = 0;
  // createdDate: Date = null;
  // createdBy: string = null;
  // updatedDateDate = null;
  // updatedBy: string = null;
  // AssociateModel?: AssociateModel;
}
